lib/stronger_parameters/controller_support/permitted_parameters.rb

```diff
@@ -0,0 +1,169 @@
+require 'stronger_parameters/constraints'
+
+module StrongerParameters
+  module ControllerSupport
+    module PermittedParameters
+      def self.included(klass)
+        klass.extend ClassMethods
+        klass.before_action :permit_parameters
+      end
+
+      class PermittedParametersHash < HashWithIndifferentAccess
+        def initialize(other = nil)
+          super()
+          merge!(other) unless other.nil?
+        end
+
+        def merge!(other)
+          return self if other.nil?
+
+          other.each do |key, value|
+            value = sugar(value)
+
+            if value.is_a?(StrongerParameters::Constraint)
+              self[key] = value
+            else
+              self[key] ||= self.class.new
+              self[key].merge!(value)
+            end
+          end
+
+          self
+        end
+
+        def merge(other)
+          dup.merge!(other)
+        end
+
+        def dup
+          super.tap do |duplicate|
+            duplicate.each do |k, v|
+              duplicate[k] = if v == true
+                true
+              else
+                v.dup
+              end
+            end
+          end
+        end
+
+        def sugar(value)
+          case value
+          when Array
+            ActionController::Parameters.array(*value.map { |v| sugar(v) }) | ActionController::Parameters.nil
+          when Hash
+            constraints = value.each_with_object({}) do |(key, v), memo|
+              memo[key] = sugar(v)
+            end
+            ActionController::Parameters.map(constraints)
+          else
+            value
+          end
+        end
+      end
+
+      DEFAULT_PERMITTED = PermittedParametersHash.new(
+        controller: ActionController::Parameters.anything,
+        action: ActionController::Parameters.anything,
+        format: ActionController::Parameters.anything,
+        authenticity_token: ActionController::Parameters.string
+      )
+
+      module ClassMethods
+        def self.extended(base)
+          base.class_eval do
+            class_attribute :log_unpermitted_parameters, instance_accessor: false
+          end
+        end
+
+        def log_unpermitted_parameters!
+          self.log_unpermitted_parameters = true
+        end
+
+        def permitted_parameters(action, permitted)
+          if permitted_parameters_list[action] == :anything && permitted != :anything
```

Comment:
should be an active verb -> `permit_parameters` ?
(would kind of collide with the before filter ... but I'd rather have a ugly before_filter then an ugly user facing method)